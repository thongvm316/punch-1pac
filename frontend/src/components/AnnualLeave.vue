<template>
  <div>
    <button type="button" class="btn btn-notice mr-5" @click="toggleAddModal">
      {{ $t('header.btnAnnualLeave') }}
    </button>

    <modal :title="$t('annualLeave.title')" :modal-open.sync="isAddModalOpen">
      <annual-leave-form v-if="isAddModalOpen" @finishRequest="isAddModalOpen = false"/>
    </modal>
  </div>
</template>

<script>
import modal from '../mixins/modal'
import AnnualLeaveForm from '../components/AnnualLeaveForm'

export default {
  name: 'annual-leave',

  components: {
    AnnualLeaveForm
  },

  mixins: [modal]
}
</script>
