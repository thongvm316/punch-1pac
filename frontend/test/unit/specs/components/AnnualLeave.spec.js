import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import AnnualLeave from '@/components/AnnualLeave'

const toggleAddModal = jest.spyOn(AnnualLeave.mixins[0].methods, 'toggleAddModal')

Object.assign(wrapperOps, {
  stubs: {
    Modal: true,
    AnnualLeaveForm: true
  }
})

describe('AnnualLeave.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(AnnualLeave, wrapperOps)
  })

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when AnnualLeave was mounted', () => {
    it('should render correctly', () => {
      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when click toggle add modal', () => {
    it('should call method', async () => {
      wrapper.find('button').trigger('click')
      await wrapper.vm.$nextTick

      expect(toggleAddModal).toHaveBeenCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
