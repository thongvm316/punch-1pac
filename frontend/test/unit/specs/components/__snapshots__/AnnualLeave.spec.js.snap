// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnualLeave.vue when AnnualLeave was mounted should render correctly 1`] = `
<div><button type="button" class="btn btn-notice mr-5">
    Day Off
  </button>
  <modal-stub title="Request a leave">
    <!---->
  </modal-stub>
</div>
`;

exports[`AnnualLeave.vue when click toggle add modal should call method 1`] = `
<div><button type="button" class="btn btn-notice mr-5">
    Day Off
  </button>
  <modal-stub title="Request a leave" modal-open="true">
    <annualleaveform-stub></annualleaveform-stub>
  </modal-stub>
</div>
`;
